pennylane/devices/default_mixed.py

```diff
@@ -0,0 +1,280 @@
+            return operation.eigvals
+
+        if isinstance(operation, Channel):
+            return operation.kraus_matrices
+
+        return [operation.matrix]
+
+    def _apply_channel(self, kraus, wires):
+        r"""Apply a quantum channel specified by a list of Kraus operators to subsystems of the
+        quantum state. For a unitary gate, there is a single Kraus operator.
+
+        Args:
+            matrices (array): Kraus operators
+            wires (Wires): target wires
+        """
+
+        channel_wires = self.map_wires(wires)
+
+        # Computes K^\dagger , which are needed for the transformation K \rho K^\dagger
+        kraus_dagger = [self._conj(self._transpose(k)) for k in kraus]
+
+        # Changes tensor shape
+        kraus_shape = len(kraus) + [2] * len(channel_wires) * 2
+        kraus = self._cast(self._reshape(kraus, kraus_shape), dtype=self.C_DTYPE)
+        kraus_dagger = self._cast(self._reshape(kraus_dagger, kraus_shape), dtype=self.C_DTYPE)
+
+        # Tensor indices of the state. For each qubit, need an index for rows *and* columns
+        state_indices = ABC[: 2 * self.num_wires]
+
+        # row indices of the quantum state affected by this operation
+        row_wires_list = channel_wires.tolist()
+        row_indices = "".join(ABC_ARRAY[row_wires_list].tolist())
+
+        # column indices are shifted by the number of wires
+        col_wires_list = [w + self.num_wires for w in row_wires_list]
+        col_indices = "".join(ABC_ARRAY[col_wires_list].tolist())
+
+        # indices in einsum must be replaced with new ones
+        new_row_indices = ABC[2 * self.num_wires : 2 * self.num_wires + len(channel_wires)]
+        new_col_indices = ABC[
+            2 * self.num_wires + len(channel_wires) : 2 * (self.num_wires + len(channel_wires))
+        ]
+
+        # index for summation over Kraus operators
+        kraus_index = ABC[
+            2 * (self.num_wires + len(channel_wires)) : 2 * (self.num_wires + len(channel_wires))
+            + 1
+        ]
+
+        # new state indices replace row and column indices with new ones
+        new_state_indices = functools.reduce(
+            lambda old_string, idx_pair: old_string.replace(idx_pair[0], idx_pair[1]),
+            zip(col_indices + row_indices, new_col_indices + new_row_indices),
+            state_indices,
+        )
+
+        # index mapping for einsum, e.g., 'iga,abcdef,idh->gbchef'
+        einsum_indices = (
+            "{kraus_index}{new_row_indices}{row_indices}, {state_indices}"
+            "{kraus_index}{col_indices}{new_col_indices}->{new_state_indices}".format(
+                kraus_index=kraus_index,
+                new_col_indices=new_col_indices,
+                col_indices=col_indices,
+                state_indices=state_indices,
+                row_indices=row_indices,
+                new_row_indices=new_row_indices,
+                new_state_indices=new_state_indices,
+            )
+        )
```

Comment:
Is this definitely simpler than `tensordot`? I guess `tensordot` has the complication with shuffling the indices at the end :thinking:

Reply:
On the other hand:

* `einsum` tends to be faster than `tensordot` as it is better optimized within NumPy

* `einsum` is more powerful than `tensordot` --- it is not always clear if an `einsum` function can be re-written using `tensordot`, as `tensordot` always has an implied contraction. E.g., `tensordot` cannot make use of the `->...` part of `einsum`!

Reply:
Not saying it was simpler, but I wanted to start with one of them. Since in `default.qubit` `einsum` appeared to be faster for most operations (acting on one or two qubits), I decided to start there. Also, I'm more familiar with `einsum` :wink: 

I definitely want to eventually implement both options and benchmark them